site/content/en/docs/Installation/upgrading.md

```diff
@@ -0,0 +1,116 @@
+gradually with production load and easily rolled back if need arises.
+
+### Multiple Clusters
+
+Essentially, we want to do a [blue/green deployment](https://martinfowler.com/bliki/BlueGreenDeployment.html) to the 
+new version of Agones.  This means that we can slowly migrate our game servers from the older version's cluster (blue)
+to the new version's cluster (green), and ensure nothing surprising happens during this process.
+
+This also allows easy rollback to the previous infrastrucutre that we already know to be working in production, with
+minimal interruptions to player experience.
+
+The following are steps to implement this:
+
+1. Create a new cluster of the same size or smaller as the current cluster.
+2. Install the new version of Agones on the new cluster.
+3. Deploy the same set of Fleets and/or GameServers from the old cluster into the new cluster.
+4. With your matchmaker, start sending a small percentage of your matched players' game sessions to the new cluster.
+5. Assuming everything is working successfully on the new cluster, slowly increase the percentage of matched sessions to the new cluster, until you reach 100%.
+6. Once you are comfortable with the stability of the new cluster with the new Agones version, shut down the old cluster.
+7. Congratulations - you have now upgraded to a new version of Agones! 👍
+
+### Single Cluster
+
+If you are upgrading a single cluster, we recommend creating a maintenance window, in which your game goes offline
+for the period of your upgrade, as there will be a short period in which Agones will be non-responsive during the upgrade.
+
+### Installation with install.yaml
+
+If you installed [Agones with install.yaml]({{< relref "_index.md#install-with-yaml" >}}), then you will need to delete
+the previous installation of Agones before upgrading to the new version, as we need to remove all of Agones before installing
+the new version.
+
+1. Start your maintenance window.
+1. Delete the current set of Fleets and/or GameServers in your cluster.
+1. Make sure to delete the same version of Agones that was previously installed, for example:
+   `kubectl delete -f https://raw.githubusercontent.com/googleforgames/agones/<old-release-version>/install/yaml/install.yaml`
+1. Install Agones [with install.yaml]({{< relref "_index.md#install-with-yaml" >}}).
+1. Deploy the same set of Fleets and/or GameServers back into the cluster.
+1. Run any other tests to ensure the Agones installation is working as expected.
+1. Close your maintenance window.
+7. Congratulations - you have now upgraded to a new version of Agones! 👍
+
+### Installation with Helm
+
+Helm features capabilities for upgrading to newer versions without having to delete the older version. For details on
+how to use Helm for upgrades, see the [helm upgrade](https://helm.sh/docs/helm/#helm-upgrade) documentation.
+
+Given the above, the steps for upgrade are simpler:
```

Comment:
I'd _really_ like to break apart and reorganise the installation docs as they currently stand -- they are one big mung of things, that diverge only sometimes for random topics.

I started to dig into what you have above, because I think we should do it, but it just made me want to pull this whole thing apart more - as then i want a section on "why choose which installation method", but then I was just unravelling more and more string as I pulled on the thread.

TL;DR - can we do this later?